src/lib/interactionManager/credentialOfferFlow.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  JWTEncodable,
+} from 'jolocom-lib/js/interactionTokens/JSONWebToken'
+import { CredentialOfferRequest } from 'jolocom-lib/js/interactionTokens/credentialOfferRequest'
+import { InteractionType } from 'jolocom-lib/js/interactionTokens/types'
+import { CredentialsReceive } from 'jolocom-lib/js/interactionTokens/credentialsReceive'
+import { SignedCredentialWithMetadata } from './types'
+import { Interaction } from './interaction'
+import { CredentialOfferResponse } from 'jolocom-lib/js/interactionTokens/credentialOfferResponse'
+import { Flow } from './flow'
+import { last } from 'ramda'
+
+type ValidationErrorMap = {
+  invalidIssuer?: boolean
+  invalidSubject?: boolean
+}
+
+export type OfferWithValidity = SignedCredentialWithMetadata & {
+  validationErrors: ValidationErrorMap
+}
+
+export class CredentialOfferFlow extends Flow {
+  public credentialOfferingState: OfferWithValidity[] = []
+  public constructor(ctx: Interaction) {
+    super(ctx)
+  }
+
+  public getState() {
+    return this.credentialOfferingState
+  }
+
+  // TODO Go back to JSONWebToken<JWTEncodable> and use guard functions when casting
```

Comment:
This should be done before the merge, but not sure if it can be done cleanly, since the `handleInteractionToken` class takes `JWTEncodable` directly, which are trickier to guard.

Reply:
Why exactly are we going back to `JSONWebToken<>`? I thought the whole point was to abstract away the token on the interaction class and handle the payload directly on the flow.

Reply:
We're not going back to JSONWebToken, the comment is old. It's about doing the typecasting when calling the function (using `as`), or keeping the function signature generic (`JWTEncodable`) and running a custom type guard to assert that we have the right JWTEncodable.

Reply:
so, we just need to remove the TODO then

Reply:
This was added in 
https://github.com/jolocom/smartwallet-app/pull/1569/commits/a1eb186cd28d233497884f5a471a0ee044b9bbe

Will await further feedback before marking this as resolved.